Bottom: b8083e94852eec1a1488e8b44683f3a9bc82d884
Top:    6aaac58a9388080b78b7d9addc12475288dacc4a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 23:11:13 +0000

Refresh of work-in-progress

---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index f8ce9df..f5f225a 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -303,10 +303,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
 
   s = fgets (buf, 1024, stdin);
   if (default_string && (s == NULL || buf[0] == 0))
-    {
-      strncpy (buf, default_string);
-    }
-
+    strncpy (buf, default_string);
   return buf;
 }
 
@@ -318,13 +315,16 @@ nogui_fileselect (const char *title, const char *descr,
 		  const char *history_tag, int flags)
 {
   static char buf[1024];
+  char *s;
+
   if (default_file)
     printf ("%s [%s] : ", title, default_file);
   else
     printf ("%s : ", title);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_file)
-    strcpy (buf, default_file);
+
+  s = fgets (buf, 1024, stdin);
+  if (default_string && (s == NULL || buf[0] == 0))
+    strncpy (buf, default_string);
   return buf;
 }
